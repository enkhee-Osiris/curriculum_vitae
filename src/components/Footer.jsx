function Footer() {
  return (
    <footer>
      <p className="text-center">
        Crafted with
        <span className="text-red-600"> â™¥ </span>
        by&nbsp;
        <a href="http://github.com/enkhee-Osiris" className="text-red-600 visited:text-red-500">
          osiris
        </a>
        .
      </p>
    </footer>
  );
}

export default Footer;
