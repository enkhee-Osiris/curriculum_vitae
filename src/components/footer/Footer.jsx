import React from 'react';

const Footer = () => (
  <footer style={{ textAlign: 'center', fontSize: '1.7rem', marginBottom: '3rem' }}>
    Crafted with
    <span style={{ color: '#fa1414' }}> â™¥ </span>
    by&nbsp;
    <a
      href="http://github.com/enkhee-Osiris"
      style={{
        textDecoration: 'none',
        color: '#fa1414',
      }}
    >
      Osiris
    </a>
    .
  </footer>
);

export { Footer as default };
