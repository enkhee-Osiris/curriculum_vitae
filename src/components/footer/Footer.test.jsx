import React from 'react';
import { shallow } from 'enzyme';
import Footer from './Footer';

describe('<Footer />', () => {
  it('should be defined', () => {
    expect(Footer).toBeDefined();
  });

  it('renders correctly', () => {
    const footer = shallow(<Footer />);

    expect(footer).toMatchSnapshot();
  });

  it('renders <3', () => {
    const footer = shallow(<Footer />);

    expect(footer.find('span').text()).toEqual(' ♥ ');
  });

  it('renders text', () => {
    const footer = shallow(<Footer />);

    expect(footer.text()).toEqual('Crafted with ♥ by Osiris.');
  });
});
