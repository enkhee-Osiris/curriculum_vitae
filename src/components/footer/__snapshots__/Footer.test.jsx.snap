// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Crafted with",
        <span
          style={
                    Object {
                              "color": "#fa1414",
                            }
          }
>
           ♥ 
</span>,
        "by ",
        <a
          href="http://github.com/enkhee-Osiris"
          style={
                    Object {
                              "color": "#fa1414",
                              "textDecoration": "none",
                            }
          }
>
          Osiris
</a>,
        ".",
      ],
      "style": Object {
        "fontSize": "1.7rem",
        "marginBottom": "3rem",
        "textAlign": "center",
      },
    },
    "ref": null,
    "rendered": Array [
      "Crafted with",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " ♥ ",
          "style": Object {
            "color": "#fa1414",
          },
        },
        "ref": null,
        "rendered": " ♥ ",
        "type": "span",
      },
      "by ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Osiris",
          "href": "http://github.com/enkhee-Osiris",
          "style": Object {
            "color": "#fa1414",
            "textDecoration": "none",
          },
        },
        "ref": null,
        "rendered": "Osiris",
        "type": "a",
      },
      ".",
    ],
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Crafted with",
          <span
            style={
                        Object {
                                    "color": "#fa1414",
                                  }
            }
>
             ♥ 
</span>,
          "by ",
          <a
            href="http://github.com/enkhee-Osiris"
            style={
                        Object {
                                    "color": "#fa1414",
                                    "textDecoration": "none",
                                  }
            }
>
            Osiris
</a>,
          ".",
        ],
        "style": Object {
          "fontSize": "1.7rem",
          "marginBottom": "3rem",
          "textAlign": "center",
        },
      },
      "ref": null,
      "rendered": Array [
        "Crafted with",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " ♥ ",
            "style": Object {
              "color": "#fa1414",
            },
          },
          "ref": null,
          "rendered": " ♥ ",
          "type": "span",
        },
        "by ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Osiris",
            "href": "http://github.com/enkhee-Osiris",
            "style": Object {
              "color": "#fa1414",
              "textDecoration": "none",
            },
          },
          "ref": null,
          "rendered": "Osiris",
          "type": "a",
        },
        ".",
      ],
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
